### PR TITLE
fix: conditionally render create answer form based on auth

### LINKED ISSUE #78: client: do not render create answer form if user not authenticated

### PR DESCRIPTION
**Pull request by AaronCQL**
closes #78

### CODE REVIEW

File: client/components/questions/ViewAnswersCard.tsx [hunk 1]
```diff
@@ -46,12 +48,14 @@ const ViewAnswersCard: FC<ViewAnswersCardProp> = ({
...
-        <AnswerForm questionId={questionId} refreshAnswers={refreshAnswers} />
-      </Space>
+      {isAuthenticated ? (
+        <Space style={{ width: "100%" }} direction="vertical" size="middle">
+          <Text style={{ fontSize: "1.2rem" }} type="secondary">
+            Your Answer
+          </Text>
+          <AnswerForm questionId={questionId} refreshAnswers={refreshAnswers} />
+        </Space>
+      ) : null}
```
Comment (EugeneTeu) [inline client/components/questions/ViewAnswersCard.tsx:58]: if user toggles again while typing halfway, he will lose his answers right? then need to show a warn before confirm toggle when he editing

Comment (AaronCQL) [reply client/components/questions/ViewAnswersCard.tsx:58]: The `firebaseUser` will only be `null` (and thus `isAuthenticated` is `false`) when the user choose to log out. Do you mean show a warning before the user chooses to log out?

Comment (EugeneTeu) [reply client/components/questions/ViewAnswersCard.tsx:58]: o no i referring to will unmount the component so will lose the state info

Comment (EugeneTeu) [reply client/components/questions/ViewAnswersCard.tsx:58]: but minor issue

**Review by EugeneTeu — APPROVED** [review]

**Pull request merged**